hicommonwealth/commonwealth
# Implement pagination for members page + inline search

## Link to Issue
Closes: #3825

## Description of Changes
- Implements infinite scroll + search input for members page
- Implements search input for Manage Community members list

Implementation Details
- The `searchProfiles` endpoint is used rather than `bulkMembers` since `bulkMembers` searches across addresses, but not profiles.
- The `searchProfiles` endpoint has been updated to optionally include community roles.
- Changed the `MentionsCache` class to a more generic `TTLCache`, used by quill mentions search and Manage Community members search (we can look into standardizing this behavior via React Query)

## Test Plan
- Go to members page, confirm that pagination and search works correctly
- Go to Manage Community page, confirm that search works correctly, confirm that selected user can be upgraded to admin/moderator

## Deployment Plan
N/A

## Other Considerations
For the Manage Community page, I implemented inline search, but not pagination.

My reasoning for this is 1.) to reduce code complexity since the logic involves profiles + roles and 2.) from a UX standpoint, when searching for a specific user, you'd likely want to start by typing the username to narrow down the search, rather than scroll through hundreds/thousands of random users.
